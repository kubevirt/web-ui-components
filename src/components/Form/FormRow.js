import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Col, ControlLabel, FormGroup, HelpBlock, FieldLevelHelp } from 'patternfly-react';
import { get } from 'lodash';

import { VALIDATION_INFO_TYPE } from '../../constants';
import { prefixedId } from '../../utils';

export const FormControlLabel = ({ isRequired, title, help }) => (
  <React.Fragment>
    <ControlLabel className={isRequired ? 'required-pf' : null}>{title}</ControlLabel>
    {help && <FieldLevelHelp className="kubevirt-form-group__field-help" placement="right" content={help} />}
  </React.Fragment>
);

FormControlLabel.defaultProps = {
  isRequired: false,
  title: '',
  help: '',
};

FormControlLabel.propTypes = {
  isRequired: PropTypes.bool,
  title: PropTypes.string,
  help: PropTypes.string,
};

export const ValidationFormRow = ({
  children,
  id,
  className,
  isHidden,
  validation,
  controlSize,
  labelSize,
  textPosition,
  ...props
}) => {
  if (isHidden) {
    return null;
  }

  return (
    <FormGroup
      id={prefixedId(id, 'id-form-row')}
      validationState={validation && validation.type !== VALIDATION_INFO_TYPE ? validation.type : null}
      className={classNames('kubevirt-form-group', className)}
    >
      <Col sm={labelSize} className={textPosition}>
        <FormControlLabel {...props} />
      </Col>
      <Col sm={controlSize}>{children}</Col>
    </FormGroup>
  );
};

ValidationFormRow.defaultProps = {
  ...FormControlLabel.defaultProps,
  children: null,
  id: null,
  className: null,
  isHidden: false,
  validation: {},
  controlSize: 5,
  labelSize: 3,
  textPosition: 'text-right',
};

ValidationFormRow.propTypes = {
  ...FormControlLabel.propTypes,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  id: PropTypes.string,
  className: PropTypes.string,
  isHidden: PropTypes.bool,
  validation: PropTypes.object,
  controlSize: PropTypes.number,
  labelSize: PropTypes.number,
  textPosition: PropTypes.string,
};

export const FormRow = ({ children, validation, ...props }) => {
  const validationMessage = get(validation, 'message', null);

  return (
    <ValidationFormRow validation={validation} {...props}>
      {children}
      {validationMessage && <HelpBlock>{validationMessage}</HelpBlock>}
    </ValidationFormRow>
  );
};

FormRow.defaultProps = ValidationFormRow.defaultProps;

FormRow.propTypes = ValidationFormRow.propTypes;
