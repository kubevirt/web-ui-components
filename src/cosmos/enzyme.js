import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
