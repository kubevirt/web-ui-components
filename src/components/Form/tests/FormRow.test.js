import React from 'react';
import { shallow } from 'enzyme';

import { FormRow, ValidationFormRow, FormControlLabel } from '../FormRow';

describe('<FormRow />', () => {
  it('renders FormRow correctly', () => {
    const component = shallow(<FormRow>test</FormRow>);
    expect(component).toMatchSnapshot();
  });
  it('renders ValidationFormRow correctly', () => {
    const component = shallow(<ValidationFormRow>test</ValidationFormRow>);
    expect(component).toMatchSnapshot();
  });
  it('renders FormControlLabel correctly', () => {
    const component = shallow(<FormControlLabel title="test" />);
    expect(component).toMatchSnapshot();
  });
});
