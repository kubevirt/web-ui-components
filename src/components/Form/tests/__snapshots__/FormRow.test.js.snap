// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormRow /> renders FormControlLabel correctly 1`] = `
<Fragment>
  <ControlLabel
    bsClass="control-label"
    className={null}
    srOnly={false}
  >
    test
  </ControlLabel>
</Fragment>
`;

exports[`<FormRow /> renders FormRow correctly 1`] = `
<ValidationFormRow
  className={null}
  controlSize={5}
  help=""
  id={null}
  isHidden={false}
  isRequired={false}
  labelSize={3}
  textPosition="text-right"
  title=""
  validation={Object {}}
>
  test
</ValidationFormRow>
`;

exports[`<FormRow /> renders ValidationFormRow correctly 1`] = `
<FormGroup
  bsClass="form-group"
  className="kubevirt-form-group"
  id={null}
>
  <Col
    bsClass="col"
    className="text-right"
    componentClass="div"
    sm={3}
  >
    <FormControlLabel
      help=""
      isRequired={false}
      title=""
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    sm={5}
  >
    test
  </Col>
</FormGroup>
`;
